### ISSUE TITLE
How do I use the AnomalyCLIP to train my custom dataset

### ISSUE THREAD

**Issue by Fotonic179** [zqhang/AnomalyCLIP#22]
I want to use AnomalyCLIP to train my custom dataset. And my dataset has no pixel ground truth.

**Comment by zqhang** [comment]
Hi, you can create a folder similar to MVTec AD dataset. Then, you could generate the corresponding JSON script for your dataset by referring to the script in 'generate_dataset_json'.

**Comment by Fotonic179** [comment]
Thanks for your reply, I&nbsp;create a folder similar to MVTec AD dataset and generate the corresponding JSON script for my dataset by referring to the script in 'generate_dataset_json'. I&nbsp;commented out the code related to img_mask in 'generate_dataset_json' and 'dataset'. I also only use the image_loss to train anomalyCLIP on my dataset.&nbsp;Excuse me, am I doing that right?




------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "zqhang/AnomalyCLIP"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2024年6月9日(星期天) 晚上9:28
***@***.***&gt;;
***@***.******@***.***&gt;;
主题:&nbsp;Re: [zqhang/AnomalyCLIP] How do I use the AnomalyCLIP to train my custom dataset (Issue #22)





 
If you do not have ground truth, just use global loss to update AnomalyCLIP.

**Comment by zqhang** [comment]
Yes, it is correct.

**Comment by Fotonic179** [comment]
Thank you very much for your reply. I trained the AnomalyCLIP on my custom dataset successfully. You can close this issue.




------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "zqhang/AnomalyCLIP"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2024年6月11日(星期二) 晚上7:26
***@***.***&gt;;
***@***.******@***.***&gt;;
主题:&nbsp;Re: [zqhang/AnomalyCLIP] How do I use the AnomalyCLIP to train my custom dataset (Issue #22)





 
Yes, it is correct.

**Comment by JackHeroFly** [comment]
谢谢作者，使用mvtec数据训练模型，并在ISIC数据上进行了测试，但是异常检测区域均以热力图形式展示，请问如何得到异常区域的二值图像呢？

**Comment by watertianyi** [comment]
> 是的，正确。
Does the training data of AnomalyCLIP require a mask? I don’t have groundtruth here. Can I train it? Can I comment out the above mask for training?
